src/physics/em/detail/UrbanMscStepLimit.hh

```diff
@@ -0,0 +1,279 @@
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ */
+CELER_FUNCTION
+UrbanMscStepLimit::UrbanMscStepLimit(const UrbanMscNativeRef& shared,
+                                     const ParticleTrackView& particle,
+                                     GeoTrackView*            geometry,
+                                     const PhysicsTrackView&  physics,
+                                     const MaterialView&      material)
+    : shared_(shared)
+    , inc_energy_(particle.energy())
+    , safety_(geometry->find_safety(geometry->pos()))
+    , params_(shared.params)
+    , msc_(shared_.msc_data[material.material_id()])
+    , helper_(shared, particle, physics, material)
+{
+    CELER_EXPECT(particle.particle_id() == shared.electron_id
+                 || particle.particle_id() == shared.positron_id);
+
+    range_  = helper_.range(inc_energy_);
+    lambda_ = helper_.msc_mfp(inc_energy_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the true path length using the Urban multiple scattering model
+ * as well as the geometry path length for a given proposed physics step. The
+ * model is selected for the candiate process governing the step if the true
+ * path length is smaller than the current physics step length. However, the
+ * geometry path length will be used for the further step length competition
+ * (either with the linear or field propagator). If the geometry path length
+ * is smaller than the distance to the next volume, then the model is finally
+ * selected for the interaction by the multiple scattering.
+ */
+template<class Engine>
+CELER_FUNCTION auto UrbanMscStepLimit::operator()(Engine& rng) -> MscResult
+{
+    MscResult result;
+
+    result.phys_step = helper_.step_length();
+    result.true_path = min<real_type>(result.phys_step, range_);
+
+    // The case for a very small step or the lower limit for the linear
+    // distance that e-/e+ can travel is far from the geometry boundary
+    // NOTE: use d_over_r_mh for muons and charged hadrons
+    real_type distance = range_ * msc_.d_over_r;
+    if (result.true_path < helper_.limit_min_fix()
+        || (safety_ > 0 && distance < safety_))
+    {
+        result.is_displaced = false;
+        result.geom_path    = calc_geom_path(result.true_path);
+        return result;
+    }
+
+    // Step limitation algorithm: UseSafety (the default)
+    // TODO: add options for other algorithms (see G4MscStepLimitType.hh)
+
+    // Initialisation at the first step or at the boundary
+    real_type range_fact = params_.range_fact;
+    real_type range_init = max<real_type>(range_, lambda_);
+
+    // G4StepStatus = fGeomBoundary: step defined by a geometry boundary
+    bool on_boundary = (safety_ == 0);
+
+    if (first_ || on_boundary)
+    {
+        if (lambda_ < params_.lambda_limit)
```

Comment:
Should the operator be flipped?
```suggestion
        if (lambda_ > params_.lambda_limit)
```

Reply:
Indeed. This is a critical bug - my bad!  Fixed.